src/handlers/scheduledEventHandler.ts

```diff
@@ -1,8 +1,54 @@
+import { KVNamespace } from '@cloudflare/workers-types';
+
 import { handleConfig } from '../config/config';
+import { KEY_NAME, NAMESPACE_NAME } from '../constants';
 import { env } from '../types/global.types';
+import { generateJwt } from '../utils/generateJwt';
 
 export async function ping(env: env) {
 	const url = handleConfig(env);
 	const response = await fetch(`${url.baseUrl}/healthcheck`);
 	return response;
 }
+
+export async function callDiscordNicknameBatchUpdate(env: env) {
+	const namespace = env[NAMESPACE_NAME] as unknown as KVNamespace;
+	let lastNicknameUpdate: string | number = 0;
+	try {
+		lastNicknameUpdate = (await namespace.get(KEY_NAME)) ?? 0;
+	} catch (err) {
+		console.log('Error while fetching the timestamp for last nickname update');
+		throw err;
+	}
+
+	const url = handleConfig(env);
+	let token;
+	try {
+		token = await generateJwt(env);
+	} catch (err) {
+		console.log(`Error while generating JWT token: ${err}`);
+		throw err;
+	}
+	const response = await fetch(`${url.baseUrl}/discord-actions/nickname/status`, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${token}`,
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({
+			lastNicknameUpdate,
```

Comment:
we are not updating this variable anywhere, also why do we need this in request body?

Reply:
It didn't make sense to pass this as a query param to the batch update. Hence, passed this via body.

Reply:
This is to ensure that we only fetch all the users status that changed after the last time this cron job ran - to avoid updating the same users' nickname multiple times.